code/modules/mob/living/carbon/xenomorph/castes/shrike/abilities_shrike.dm

```diff
@@ -26,7 +26,19 @@
 	new /obj/effect/temp_visual/telekinesis(get_turf(caller))
 	caller.visible_message("<span class='xenowarning'>A strange buzzing hum starts to emanate from \the [caller]!</span>", \
 	"<span class='xenodanger'>We call forth the larvas to rise from their slumber!</span>")
-	notify_ghosts("\The <b>[caller]</b> is calling for the burrowed larvas to wake up!", enter_link = "join_larva=1", enter_text = "Join as Larva", source = caller, action = NOTIFY_JOIN_AS_LARVA)
+
+	var/burrowed = shrike_hive.stored_larva
+	var/total = 0
+	for(var/i in 1 to burrowed)
+		var/mob/M = get_alien_candidate()
+		if(!M)
+			break
+
+		shrike_hive.spawn_larva(M, src)
+		total++
+
+	if(burrowed != total)

```

Comment:
```suggestion
	if(shrike_hive.stored_larva)
		RegisterSignal(shrike_hive, list(COMSIG_HIVE_XENO_MOTHER_PRE_CHECK, COMSIG_HIVE_XENO_MOTHER_CHECK), .proc/is_burrowed_larva_host)
```

Reply:
And remove the previous `RegisterSignal` in the proc.
No need to register it if there's no burrowed after the attempt to get candidates.
Include the `addtimer` in this `if` tab.